### DISCUSSION TITLE
Running in Google Collab

### DISCUSSION THREAD

**Discussion by dcpatterson87** [kyleskom/NBA-Machine-Learning-Sports-Betting#376] [General]
Is this possible to run this entirely in google collab? I can get it to run and produce the initial outputs but none of the commands to update the data or train the data further seem to be working (they return syntax errors). 

I have tried repeatedly to get this to run natively on my machine but no matter what I do I get the "OSError: Cannot parse keras.." error, even after trying multiple different versions of python and pandas as suggested in other discussions.

Does anybody have any insight on how to get this properly running on google collab or have any suggestions on other things I can try? Any and all help is greatly appreciated!

**Comment by aaaclones** [comment]
I suggest you don't bother running this , this model isn't accurate . Its extremely misleading and well underperforming your probably better off just asking chatgpt whose going to win

**Reply by kyleskom** [reply]
What have you done to improve it or work on it? What part has been misleading, i'm not telling you to just use this and nothing else.

**Reply by aaaclones** [reply]
The statistical confidence in which the output your model has been giving that part is purely misleading when you have your xgb at one point claiming 65%+ accuracy on the totals , your ev values are nowhere close to what it claims to . What i am doing to the project is raising awareness . 90% confidence levels on some of the predictions that are nowhere near it . 90% strength of confidence level in your prediction when it's more like 90% it's probably not right . I didn't create this repository but so it's not my job to try and improve it . I tried to propose some upgrades but you and your fan boy shot it down , I tried to offer you guys api keys that have up to minute status on every single players past and present . My contribution at the moment is to make sure they don't take these data value points that your poor model is putting out serious . Your acting as if your doing something at service these people can just simply challenge your model with chatgpt if they knew how to prompt it properly . Dont get upset at me just simply improve your model . Your not telling them enough you and your loud mouth sidekick is just acting as if things are going to get better when clearly it's not

**Reply by kyleskom** [reply]
" I didn't create this repository but so it's not my job to try and improve it " Its open source, everything to contribute is here. I never claimed this to be a fool proof tool. Instead of just bitching and posting ideas, contribute change yourself instead of waiting for others to do it for you. Sure maybe chatGPT is a better tool right now, I don't know that I haven't tried, but this project is from long before chatGPT even existed.

**Reply by aaaclones** [reply]
Furthermore why don't you put up how your model is doing ? You have a ton of people in this repository that are newbies so they will take your information as if it's gospel and get rekt . Why don't you put your results up and see if your positive ev or not ? why don't you put your results up for xgb up for the totals rahter just the falsely misrepresented winning % ? So yeah that's the misleading part about your model . Your not looking for improve the model I offered your little fan boy some help he damn near lost his mind , You guys can keep this faulty broken model . It doesnt even perform better than Chatgpt or Bard so chill out with the false claims on your e/v and xgb .

**Reply by kyleskom** [reply]
I don't track every single predictions so I don't have that. Im almost certain you have no idea how models are trained and percentages are obtained. On top on that im also certain you have no idea how or what expected value is and calculated.

**Reply by aaaclones** [reply]
Its troubling and misleading you can't spew out 90% confidence levels and be dead wrong on all of them . 90% means your right 9/10 times it's like Steph Curry at the free throw line but yet your model has missed all of those and that can cause people to do some dumb stuff if they were new to the repository and saw 92% statistical confidence. You asked what was misleading and I am telling you . You and your little fanboy don't want help with this project , I wont let people lose any money by following these predictions . So since you won't tell them I will , Machine learning with human feed back so far your feed back has been shitty at best so how about you let me try and give the feedback while you bash your head trying to upgrade this model . Or simply block me if you feel i am the one giving the misinformation about your model . Bottom line is your model is not accurate if i am lying just block me .

**Reply by aaaclones** [reply]
Ev = expected value , This repository = - Ev . How do I  not know how your model is trained I see it you have it in. your files ? It's data sets from the past 10 years done in a tensorflow library over the columns of data points you fitted . wtf do you mean by that ?

**Reply by kyleskom** [reply]
Im not going to block you over speaking your mind, you have your right to. I think your inaccurate in somethings your saying thats all. I don't market this as an end all be all tool, I don't agree the %'s are misleading based on using test train split of the data. I see your concern with the confidence percentages or how tools like chatGPT may be better now.

Like I said this is an open source project that I started to learn ML many years ago, this was never supposed to "blow up" the way it has. It's here for people to improve upon. It's free for everyone and something anyone can work on.

You keep mentioning my fan boy like anyone else speaks for me or represents anything.

**Reply by kyleskom** [reply]
A few last points. It's been known this model doesn't account for players. I haven't followed the NBA as close this year, I don't know about injuries ,  trades, etc that happen that cause changes in predictions the model could never know. Iv have brought up this shortcoming many times. As for ev, ev is a calculation that includes game lines as well, I cant false claim the ev when its literally a math equation.

**Reply by aaaclones** [reply]
Now that the repository has gone full nuclear there should be a disclaimer saying it's not accurate . Since you seem to have a problem with human feedback than you can leave that part up to me I will inform them unless you delete my comments otherwise I will inform them and make sure to tell them you don't look at the predictions you dont bother to see how accurate it is so they need to make sure to not take the information as gospel .

**Reply by kyleskom** [reply]
Go for it, looks like you have far my time on your hands then I do.

**Reply by aaaclones** [reply]
The ev claims are only true when your information is true . But since you don't have player updates or injuries it skews the data you have predictions in some games with 2 key star players out as if they are still in most certainly this will create a delta discrepancy wouldn't it ? Your model makes predictions solely on team stats not player stats so if player A , B  , C is out it doesn't account for that it predicts as tho the team stats aren't skewed that creates a breach in the team data . Its like saying let me predict as tho Michael Jordan is playing when he's not and give the same data points for the team ???

**Reply by kyleskom** [reply]
Yes excatly and iv been saying this is a issue with this model. Iv always said treat this as a tool in a toolbelt.

**Reply by albeng8** [reply]
@aaaclones  someone did track some results from 2022-2023 season : https://docs.google.com/spreadsheets/d/1LxPq1ZF3k6BreB8p50vESP-7BTZXVrFs6abU7qxaq2c/edit#gid=1616360111 . 

I am not sure how you are using chatgpt to do a prediction when you haven't backtested it . I did some testing on chatgpt using some advanced data from nba stats and you simply cant do it due to resource constrained. I did used chatgpt to hep me with some code and run it on my local computer and it does work that way. 

I looked at how they backtest it and it was based on last 10 years vs closing line . If this model has an accuracy of 54% , it is beating the vegas odd with vig. 

I do agree with the weirdness of the  confidence level that the model churn out . I am also seeing some random confidence when i ran on my own custom dataset. 

does this model account for injury or lineup change ? Yes and no. If lebron is out for the whole season , the team stat should reflect that . This model will not account for short term injury / lineup changes so you will need to make manual adjustment to the output.

**Reply by aaaclones** [reply]
For the year the model is getting crushed . The model doesn't account for anything outside the data points in the columns it's being ran on . Kyle's little sidekick always seems to think the model is being over fitted which is pure comedy . So no the model doesn't account for lineup change and to your question sure if lebron is out for the season the team stat will reflect it your right but what if he's out for just that game. ? or what if it's not lebron whose out but just Anthony davis or player 3 ? this model doesn't account for any of that .. not only that it spews out a statistical confidence level in the 90% and 80% meanwhile it's getting smashed to smithereens . If you prompt chatgpt properly with the proper scripts it will put this shitty model to shame .

**Reply by aaaclones** [reply]
please note at 53% sure your turning a small profit but is its way to volatile and stressful to deal with at 53% to comfortably say your beating it you would need to be somewhere in the 58% or better to be sitting comfortable and feeling you have the best of it . Now if your beating the lines and coming back on the number then sure you will have way the best of it but the books will never let you keep beating the number and middling the line . For example if you bet over 225 and now the number is 227 and you bet under 227 you will lose vig on one side and for the year the games land in between the lines roughly 50-60 times so if you keep beating the number and coming back on it with a hedge you will lose vig one way but be +50 units on the low end most certainly you will feel good about a bet you either scoop or lose vig  . Goodluck to you man I advise you just to use the machine learning script Kyle has to managed to share he does deserve a lot of credit for being open source  . The only value this repository has is the framework in the machine learning it provided but outside of that please dont follow the information to bet and don't let his little sidekick convince you otherwise he will claim over fitting and underfitting . Theres machine learning models that compiles and process 1000's of data points to come up with a prediction a far more accurate one than sports outcomes .

**Reply by albeng8** [reply]
I dont think you understand how efficient the nba market is . Getting 1-2% edge over closing with vig is crushing . If you can come up with 58% model and you are able to put down money to the book without getting limited , props to you.

Your example with the line movement clearly show you dont understand how sport betting work.  I use market data + the output of my models to give me an idea which side i should bet.  Yes , if the line move against me , it either means that there are some new information came out such as  last minute lineup changes / injury.

Please dont discredit the effort Kyle put into this. It is a pretty good starter model and with tuning with more advanced data , it can probably get maybe .5-.1% more accurate.

**Comment by aaaclones** [comment]
stfu you don't understand how it works , at 1% edge you think it's good
enough to keep putting your money on 1% and riding it ? go ahead and do it
then you dummy don't try and start a conversation being condescending you
obviously have no clue on how to beat sports otherwise you would know this
repository is trash , if the line moves against you chances are your on the
wrong side and you can't middle . Go fan out with Kyle you dummy . Get lost
and dont talk to me anymore your not worthy

**Reply by albeng8** [reply]
good luck with your 58% model and get your money down to bet.

**Comment by Emillzz45** [comment]
Buddy said “beat sports betting” 😂 You a sharp or something?

**Comment by aaaclones** [comment]
your dumbass said 1% is crushing lmao stfu get out of here that's not how it works go push marginal edges and think your beating it

**Reply by albeng8** [reply]
good luck with your 5% edge model.
